stackslib/src/chainstate/stacks/boot/mod.rs

```diff
@@ -575,6 +603,63 @@ impl StacksChainState {
         .map(|value| value.expect_bool())
     }
 
+    pub fn make_signer_set(
+        threshold: u128,
+        entries: &[RawRewardSetEntry],
+    ) -> Option<Vec<NakamotoSignerEntry>> {
+        let Some(first_entry) = entries.first() else {
+            // entries is empty: there's no signer set
+            return None;
+        };
+        // signing keys must be all-or-nothing in the reward set
+        let expects_signing_keys = first_entry.signing_key.is_some();
+        for entry in entries.iter() {
+            if entry.signing_key.is_some() != expects_signing_keys {
+                panic!("FATAL: stacking-set contains mismatched entries with and without signing keys.");
+            }
+        }
+        if !expects_signing_keys {
+            return None;
+        }
+
+        let mut signer_set = BTreeMap::new();
+        for entry in entries.iter() {
+            let signing_key = if let Some(PrincipalData::Standard(s)) = entry.signing_key.clone() {
+                StacksAddress::from(s)
+            } else {
+                // TODO: should figure out if in mainnet?
```

Comment:
Yes

Reply:
I'd argue this is purely a code-aesthetic choice, really. This branch is only reached if the `signing_principal` set in the pox-4 contract is a contract principal, and the important thing is just that the address is unusable, which the burn address will be regardless of version bit.

Reply:
This is actually moot now that we know from @jferrant that we need the keys themselves.